backend/src/ml_space_lambda/authorizer/lambda_function.py

```diff
@@ -251,9 +251,10 @@ def lambda_handler(event, context):
                                 if project_user:
                                     policy_statement["Effect"] = "Allow"
                 elif "groupName" in path_params:
+                    is_group_member = _is_group_member(path_params["groupName"], username)
                     if IS_ADMIN and request_method in ["POST", "PUT", "DELETE"]:
                         policy_statement["Effect"] = "Allow"
-                    elif request_method == "GET":
+                    elif request_method == "GET" and (IS_ADMIN or is_group_member):
```

Comment:
How would you feel about simplifying this at this point? It looks like if they're an admin, they get a 👍🏻 so then we can remove some of that clutter of checking the method in that scenario. Then we don't have to worry about admins when checking if they're a member.

```python
is_group_member = _is_group_member(path_params["groupName"], username)
if IS_ADMIN:
  policy_statement["Effect"] = "Allow"
elif request_method == "GET" and is_group_member:
  policy_statement["Effect"] = "Allow"
```